ccl/changefeedccl: TestChangefeedPropagatesTerminalError flaky

This test flaked on #95000: https://teamcity.cockroachdb.com/viewLog.html?buildId=8270138&tab=buildResultsDiv&buildTypeId=Cockroach_UnitTests_BazelUnitTests

```
=== RUN   TestChangefeedPropagatesTerminalError/many_aggregators
    helpers_test.go:837: making kafka feed factory
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:260
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:835
*   | github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Cancel
*   | 	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:527
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).setupFlows.func5
*   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:590
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470
*   | runtime.goexit
*   | 	GOROOT/src/runtime/asm_amd64.s:1594
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1a8 pc=0x2439bd2]

goroutine 28568 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x32f3a71?, {0x6d50bd8, 0xc00755e7b0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x6a
panic({0x4fac820, 0x9ae25f0})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Cancel(0xc004c2e700?)
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:527 +0x12
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).setupFlows.func5({0xc001a67000?, 0x0?})
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:590 +0x1d5
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:461 +0x43b
```


Jira issue: CRDB-23299

Epic CRDB-11732

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -6008,7 +6008,6 @@ func TestChangefeedHandlesDrainingNodes(t *testing.T) {
 
 func TestChangefeedPropagatesTerminalError(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 95057, "flaky test")
 	defer log.Scope(t).Close(t)
 
 	opts := makeOptions()

